com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -88,14 +103,16 @@ internal void __endSendServerRpc(NetworkSerializer serializer, uint rpcMethodId,
                 return;
             }
 
-            var rpcQueueContainer = NetworkManager.RpcQueueContainer;
+            SetUpdateStage(ref serverRpcParams.Send);
+
+            var messageQueueContainer = NetworkManager.MessageQueueContainer;
             if (IsHost)
             {
-                rpcQueueContainer.EndAddQueueItemToFrame(serializer.Writer, RpcQueueHistoryFrame.QueueFrameType.Inbound, serverRpcParams.Send.UpdateStage);
+                messageQueueContainer.EndAddQueueItemToFrame(serializer.Writer, MessageQueueHistoryFrame.QueueFrameType.Inbound, serverRpcParams.Send.UpdateStage);
```

Comment:
So, for the netstats metrics for profiling, we've modified the signature of `EndAddQueueItemToFrame` to return the RPC message size (see https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/commit/8b56bd60df12489b7e72086bbf356b8ef85869c8). If it could be doable to do that here so we can use it, it would be great. That was approved by @NoelStephensUnity . Of course, we can always do it ourselves if you can't.

Reply:
That update should possibly be a separate PR, as this PR's primary focus is around message ordered.  However, the solution to return the actual size of the RPC itself was most definitely the best idea when it comes to tracking RPC message size metrics.

Reply:
Agreed, that's outside the scope of this PR, I think.